samples/ClassroomLabs/Modules/Library/Tools/LabCreationLibrary.psm1

```diff
@@ -0,0 +1,598 @@
+                param($path)
+
+                Set-StrictMode -Version Latest
+                $ErrorActionPreference = 'Stop'
+
+                $modulePath = Join-Path $path '..\Az.LabServices.psm1'
+                Import-Module $modulePath
+                # Really?? It got to be the lines below? Doing a ForEach doesn't work ...
+                $input.movenext() | Out-Null
+            
+                $obj = $input.current[0]
+
+                Write-Verbose "object inside the newazmultiple block $obj"
+                $obj | New-AzLabSingle
+            }
+
+            Write-Host "Starting creation of all labs in parallel. Can take a while."
+            $jobs = @()
+
+            $ConfigObject | ForEach-Object {
+                Write-Verbose "From config: $_"
+                $jobs += Start-ThreadJob  -InitializationScript $init -ScriptBlock $block -ArgumentList $PSScriptRoot -InputObject $_ -Name $_.LabName -ThrottleLimit $ThrottleLimit
+            }
+
+            while (($jobs | Measure-Object).Count -gt 0) {
+                $completedJobs = $jobs | Where-Object {($_.State -ieq "Completed") -or ($_.State -ieq "Failed")}
+                if (($completedJobs | Measure-Object).Count -gt 0) {
+                    # Write output for completed jobs, but one by one so output doesn't bleed 
+                    # together, also use "Continue" so we write the error but don't end the outer script
+                    $completedJobs | ForEach-Object {
+                        $_ | Receive-Job -ErrorAction Continue
+                    }
+                    # Trim off the completed jobs from our list of jobs
+                    $jobs = $jobs | Where-Object {$_.Id -notin $completedJobs.Id}
+                    # Remove the completed jobs from memory
+                    $completedJobs | Remove-Job
+                }
+                # Wait for 60 sec before checking job status again
+                Start-Sleep -Seconds 60
+            }
+        }
+
+
+        # Needs to create resources in this order, aka parallelize in these three groups, otherwise we get contentions:
+        # i.e. different jobs trying to create the same common resource (RG or lab account)
+        New-ResourceGroups  -ConfigObject $aggregateLabs
+        New-Accounts        -ConfigObject $aggregateLabs
+        New-AzLabMultiple   -ConfigObject $aggregateLabs
+    }
+}
+
+function Set-LabProperty {
+    [CmdletBinding()]
+    param(
+        [parameter(Mandatory = $true, HelpMessage = "Array containing one line for each lab to be created", ValueFromPipeline = $true)]
+        [ValidateNotNullOrEmpty()]
+        [psobject[]]
+        $labs,
+
+        [Parameter(Mandatory = $true, ValueFromRemainingArguments=$true, HelpMessage = "Series of multiple -propertyName propValue pairs")]
+        [ValidateNotNullOrEmpty()]
+        [string[]]
+        $vars
+    )
+    begin {
+        #Convert vars to hashtable
+        $htvars = @{}
+        $vars | ForEach-Object {
```

Comment:
Should we have some validation on exactly which properties are allowed to be changed?  Not all of them are changable after the lab has been created.  The user could also be passing something like "provisioningState" to change, or the Lab Name, and those aren't changable (as far as I know)...